src/algorithm/inverover.cpp

```diff
@@ -56,245 +55,228 @@ inverover::inverover(int gen, double ri, initialization_type ini_type)
-		        break;
-		    case problem::base_tsp::CITIES:
-		        fitness[i] = prob->objfun(my_pop[i]);
-		        break;
+	std::vector<fitness_vector>  fitness(NP, fitness_vector(1));
+	for(size_t i=0; i < NP; i++){
+		switch( prob->get_encoding() ) {
+			case problem::base_tsp::FULL:
+				fitness[i] = prob->objfun(prob->full2cities(my_pop[i]));
+				break;
+			case problem::base_tsp::RANDOMKEYS:
+				fitness[i] = prob->objfun(prob->cities2randomkeys(my_pop[i], pop.get_individual(i).cur_x));
+				break;
+			case problem::base_tsp::CITIES:
+				fitness[i] = prob->objfun(my_pop[i]);
+				break;
 		}
-}
+	}
 
 
-decision_vector tmp_tour(Nv);
-bool stop, changed;
-size_t rnd_num, i2, pos1_c1, pos1_c2, pos2_c1, pos2_c2; //pos2_c1 denotes the position of city1 in parent2
-fitness_vector fitness_tmp;
+	decision_vector tmp_tour(Nv);
+	bool stop, changed;
+	size_t rnd_num, i2, pos1_c1, pos1_c2, pos2_c1, pos2_c2; //pos2_c1 denotes the position of city1 in parent2
+	fitness_vector fitness_tmp;
 
-//InverOver main loop
-for(int iter = 0; iter < m_gen; iter++){
-	for(size_t i1 = 0; i1 < NP; i1++){
-		tmp_tour = my_pop[i1];
-		pos1_c1 = unif_Nv();
-		stop = false;
-		changed = false;
-		while(!stop){
-			if(unif_01() < m_ri){
-				rnd_num = unif_Nvless1();
-				pos1_c2 = (rnd_num == pos1_c1? Nv-1:rnd_num);
-			}
-			else{
-				i2 = unif_NPless1();
-				i2 = (i2 == i1? NP-1:i2);
-				pos2_c1 = std::find(my_pop[i2].begin(),my_pop[i2].end(),tmp_tour[pos1_c1])-my_pop[i2].begin();
-				pos2_c2 = (pos2_c1 == Nv-1? 0:pos2_c1+1);
-				pos1_c2 = std::find(tmp_tour.begin(),tmp_tour.end(),my_pop[i2][pos2_c2])-tmp_tour.begin();
-			}
-			stop = (abs(pos1_c1-pos1_c2)==1 || abs(pos1_c1-pos1_c2)==Nv-1);
-			if(!stop){
-				changed = true;
-				if(pos1_c1<pos1_c2){
-					for(size_t l=0; l < (double (pos1_c2-pos1_c1-1)/2); l++){
-						std::swap(tmp_tour[pos1_c1+1+l],tmp_tour[pos1_c2-l]);}
-				pos1_c1 = pos1_c2;
+	//InverOver main loop
+	for(int iter = 0; iter < m_gen; iter++) {
+		for(size_t i1 = 0; i1 < NP; i1++) {
+			tmp_tour = my_pop[i1];
+			pos1_c1 = unif_Nv();
+			stop = false;
+			changed = false;
+			while(!stop){
+				if(unif_01() < m_ri) {
+					rnd_num = unif_Nvless1();
+					pos1_c2 = (rnd_num == pos1_c1? Nv-1:rnd_num);
+				} else {
+					i2 = unif_NPless1();
+					i2 = (i2 == i1? NP-1:i2);
+					pos2_c1 = std::find(my_pop[i2].begin(),my_pop[i2].end(),tmp_tour[pos1_c1])-my_pop[i2].begin();
+					pos2_c2 = (pos2_c1 == Nv-1? 0:pos2_c1+1);
+					pos1_c2 = std::find(tmp_tour.begin(),tmp_tour.end(),my_pop[i2][pos2_c2])-tmp_tour.begin();
 				}
-				else{
-					//inverts the section from c1 to c2 (see documentation Note3)
-					for(size_t l=0; l < (double (pos1_c1-pos1_c2-1)/2); l++){
-						std::swap(tmp_tour[pos1_c2+l],tmp_tour[pos1_c1-l-1]);}	
-				pos1_c1 = (pos1_c2 == 0? Nv-1:pos1_c2-1);				
+				stop = (abs(pos1_c1-pos1_c2)==1 || static_cast<problem::base::size_type>(abs(pos1_c1-pos1_c2))==Nv-1);

```

Comment:
This is the only functional difference:
```c++
abs(...) == Nv-1
```
```c++
static_cast<problem::base::size_type>(abs(...)) == Nv-1
```

I don't think it can break anything ? @ingmar-getzner